src/Instrumentation/Psr15/src/Psr15Instrumentation.php

```diff
@@ -0,0 +1,110 @@
+                $span = $instrumentation->tracer()->spanBuilder(sprintf('%s::%s', $class, $function))
+                    ->setAttribute('code.function', $function)
+                    ->setAttribute('code.namespace', $class)
+                    ->setAttribute('code.filepath', $filename)
+                    ->setAttribute('code.lineno', $lineno)
+                    ->startSpan();
+
+                Context::storage()->attach($span->storeInContext(Context::getCurrent()));
+            },
+            static function (MiddlewareInterface $middleware, array $params, ?ResponseInterface $response, ?Throwable $exception) {
+                $scope = Context::storage()->scope();
+                $scope?->detach();
+                if (!$scope) {
+                    return;
+                }
+                $span = Span::fromContext($scope->context());
+                if ($exception) {
+                    $span->recordException($exception, [TraceAttributes::EXCEPTION_ESCAPED => true]);
+                    $span->setStatus(StatusCode::STATUS_ERROR, $exception->getMessage());
+                }
+                $span->end();
+            }
+        );
+
+        /**
+         * Create a span to wrap RequestHandlerInterface::handle
+         * Stores the span as a request attribute which may be accessed by later hooks.
+         */
+        hook(
+            RequestHandlerInterface::class,
+            'handle',
+            static function (RequestHandlerInterface $handler, array $params, string $class, string $function, ?string $filename, ?int $lineno) use ($instrumentation) {
+                $request = ($params[0] instanceof ServerRequestInterface) ? $params[0] : null;
+                $parent = Context::getCurrent();
+                if ($request instanceof ServerRequestInterface && !$request->getAttribute(SpanInterface::class)) {
+                    //extract from request headers
+                    $parent = (new TraceContextPropagator())->extract($request->getHeaders());
+                }
+                $span = $instrumentation->tracer()->spanBuilder(sprintf('HTTP %s', $request?->getMethod() ?? 'unknown'))
+                    ->setParent($parent)
+                    ->setAttribute('code.function', $function)
+                    ->setAttribute('code.namespace', $class)
+                    ->setAttribute('code.filepath', $filename)
+                    ->setAttribute('code.lineno', $lineno)
+                    ->setAttribute(TraceAttributes::HTTP_URL, (string) $request?->getUri())
+                    ->setAttribute(TraceAttributes::HTTP_METHOD, $request?->getMethod())
+                    ->setAttribute(TraceAttributes::HTTP_REQUEST_CONTENT_LENGTH, $request?->getHeaderLine('Content-Length'))
+                    ->setAttribute(TraceAttributes::HTTP_SCHEME, $request?->getUri()?->getScheme())
+                    ->startSpan();
+
+                Context::storage()->attach($span->storeInContext(Context::getCurrent()));
```

Comment:
```suggestion
                Context::storage()->attach($span->storeInContext($parent));
```
Otherwise we lose extracted baggage etc.

Reply:
done